src/Discord/Parts/User/Member.php

```diff
@@ -170,44 +170,63 @@ public function moveMember($channel)
      *
      * @param Role|int $role The role to add to the member.
      *
-     * @return bool Whether adding the role succeeded.
+     * @return \React\Promise\Promise
      */
     public function addRole($role)
     {
+        $deferred = new Deferred();
         if ($role instanceof Role) {
             $role = $role->id;
         }
 
         // We don't want a double up on roles
         if (false !== array_search($role, (array) $this->attributes['roles'])) {
-            return false;
+            $deferred->reject();
+            return $deferred->promise();
         }
 
         $this->attributes['roles'][] = $role;
 
-        return true;
+
+        $this->http->put(
+            "guilds/{$this->guild_id}/members/{$this->id}/roles/{$role}"
+        )->then(function ($resp) use ($deferred) {
```

Comment:
Here you can just use `\React\Partial\bind([$deferred, 'resolve'])` like you do below, it is cleaner.